CMakeLists.txt

```diff
@@ -2476,29 +2480,27 @@ if(MAD)
 endif()
 
 # Media Foundation AAC Decoder Plugin
-if(WIN32)
-  find_package(MediaFoundation)
-  cmake_dependent_option(MEDIAFOUNDATION "Media Foundation AAC decoder plugin" ON "MediaFoundation_FOUND" OFF)
-  if(MEDIAFOUNDATION)
-    if(NOT MediaFoundation_FOUND)
-        message(FATAL_ERROR "MediaFoundation support requires the MediaFoundation libraries and its development headers.")
-    endif()
-    target_sources(mixxx-lib PRIVATE
-      src/sources/soundsourcemediafoundation.cpp
-    )
-    target_compile_definitions(mixxx-lib PUBLIC __MEDIAFOUNDATION__)
-    target_include_directories(mixxx-lib SYSTEM PRIVATE
-      ${MediaFoundation_INCLUDE_DIRS}
-    )
-    target_link_libraries(mixxx-lib PUBLIC
-      ${MediaFoundation_LIBRARIES}
-    )
-  endif()
+#
+# The Media Foundtation API is only available on Windows, therefore this option
+# is forcibly set off on all other platforms.
+cmake_dependent_option(MEDIAFOUNDATION "Media Foundation AAC decoder plugin" ON "WIN32" OFF)
```

Comment:
Why not default_option()?

Reply:
Same case as for core Audio. I don't care what we use, but it might be less confusion for cmake-GUI users to not show this option because media foundation only exists on windows.

Reply:
Sure.

Maybe we should add a note to the default_option() macro that explains the different behavior compared to cmake_dependent_option()?

Reply:
https://github.com/mixxxdj/mixxx/pull/3720/commits/6c4fa4055d1acea5f28c9e1c3ace810b7ac8266e